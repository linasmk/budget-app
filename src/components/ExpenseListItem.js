/* ========= App Dependencies ============= */
import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

/* ========= Code ============= */

const ExpenseListItem = ({ id, description, note, amount, createdAt }) => (
  <div className="expens-item">
    <h3 className="expense-item__description">
      <Link className="expens-item__link" to={`/edit/${id}`}>
        {description}
      </Link>
    </h3>
    <p className="expense-item__date">
      {moment(createdAt).format("MMMM Do, YYYY. [At:] LT")}
    </p>
    <p className="expense-item__amount">
      {"â‚¬" + numeral(amount / 100).format("0,0.00")}
    </p>
    <p className="expense-item__note">{note}</p>
  </div>
);

export default ExpenseListItem;
