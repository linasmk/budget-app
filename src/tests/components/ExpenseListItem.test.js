/* ========= App Dependencies ============= */
import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";
/* ========= Code ============= */

test("Should render ExpenseListItem correctly", () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});
