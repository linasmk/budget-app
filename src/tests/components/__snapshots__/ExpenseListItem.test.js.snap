// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseListItem correctly 1`] = `
<section>
  <h3>
    <Link
      to="/edit/1"
    >
      Ice cream
    </Link>
  </h3>
  <p>
    â‚¬25.55
    -
    January 1st, 1970. At: 1:00 AM
  </p>
  <p />
</section>
`;
